.pre-commit-config.yaml

```
@@ -1,13 +1,30 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
+exclude: 'tools/schemacode/bidsschematools/tests/data/broken_dataset_description.json'
 repos:
-  - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.0.1
-    hooks:
-      - id: trailing-whitespace
-      - id: end-of-file-fixer
-      - id: check-yaml
-      - id: check-json
-      - id: check-ast
-      - id: check-added-large-files
-      - id: check-case-conflict
+
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.3.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-json
+    -   id: check-ast
+    -   id: check-added-large-files
+    -   id: check-case-conflict
+
+-   repo: https://github.com/psf/black
+    rev: 22.6.0
+    hooks:
+    -   id: black
+
+-   repo: https://github.com/pycqa/flake8
+    rev: '4.0.1'
```

should we use the most recent version?

```suggestion
    rev: '5.0.4'
```